### PR TITLE
Fix: verify touch gestures in adapt, audit, and harden (#805)

### LINKED ISSUE #805: [Bug] adapt/audit verification can miss broken touch interactions
## What happened?

After running Impeccable `adapt`, `critique`, and `audit` passes on the Impeccable website, the homepage before/after slider still could not be dragged on iOS. This was discovered through manual use rather than reported by those passes.

Follow-up investigation found related gaps in other controls: faders intercepted sideways scrolling, a second pointer could redirect an adjustment, and a small sideways drift during vertical scrolling could select another tab. A subsequent code review also found that losing window focus during a mouse/pen drag could leave tab scrubbing stuck until reload.

The iOS slider is the original reported miss. The later findings demonstrate useful verification scenarios; they are not evidence that every earlier command independently missed every bug. The website fixes have since been merged.

## Steps to reproduce

Historical sequence, rather than a deterministic reproduction of an LLM run:

1. Run `adapt`, `critique`, and `audit` against a page containing custom sliders and draggable controls.
2. Review the findings, then exercise the before/after slider with touch on iOS.
3. Observe that a control can look correct at mobile widths while its primary touch interaction fails.

The earlier run traces and exact command arguments are not included, so this report does not establish which verification steps those runs performed or skipped.

## Expected behavior

`adapt` should exercise the primary touch behavior of in-scope controls, or explicitly report that it was not verified. `audit` should check interaction correctness alongside layout and accessibility. `harden` should cover interrupted gestures and recovery. Users should not need to know pointer-event implementation details to request this coverage.

Add a shared, bounded verification checklist to the relevant command references:

- Identify the custom sliders, drag surfaces, and scrollable control strips in scope.
- Exercise taps and drags with the relevant input methods; verify that scrolling across a control does not accidentally activate it or prevent intended page/container scrolling.
- Where applicable, check secondary pointers, cancellation, lost capture, release outside the control, and window focus loss. Verify that a fresh interaction works afterward.
- Use available browser/input tooling and distinguish viewport emulation, actual touch-input testing, browser-engine coverage, and physical-device testing. Explicitly report unavailable or untested coverage; do not require inaccessible hardware or treat screenshots as touch verification.
- Report which scenarios were exercised and add behavioral regressions for confirmed fixes where practical.

## Provider & environment

- Original failure: mobile web on iOS; exact device, OS version, and earlier provider version were not recorded here.
- Guidance reviewed: installed Impeccable 4.3.1, plus current upstream command references.
- The guidance improvement applies across providers.

## Additional context

- [`adapt`](https://github.com/pbakaus/impeccable/blob/main/skill/reference/adapt.md#L179) already requires real-device, Safari/iOS, and touch testing, but does not spell out custom-control gesture scenarios.
- [`audit`](https://github.com/pbakaus/impeccable/blob/main/skill/reference/audit.md#L46) focuses its responsive checklist on widths, target sizes, overflow, text scaling, and breakpoints.
- [`harden`](https://github.com/pbakaus/impeccable/blob/main/skill/reference/harden.md#L292) includes general manual/E2E testing, without an explicit gesture interruption/recovery checklist.

This is a request to strengthen verification in the existing commands, without adding another command or promising exhaustive bug detection.

AI-assisted issue preparation and source review with OpenAI Codex, at the maintainer's request.

### PR DESCRIPTION
**Pull request by abdulwahabone**
The adapt, audit, and harden references listed devices, browsers, and layout properties in their verification sections but never had the agent exercise a control's primary gesture, so an emulated viewport plus screenshots could pass as touch testing (the iOS before/after slider miss). adapt now exercises the primary gesture and the scroll-across trade and reports what produced the evidence; audit checks broken touch interaction with code-level tells; harden covers interrupted gestures and recovery. A reference-contract test pins the three sections.

Fixes #805

Validation: `bun run build` (both prose gates clean) and the full default `bun run test` against an engine built from this branch's source (core, oracle with 832 cases, detector, live, framework, plugin-e2e all green). Generated provider output intentionally omitted.

Prepared with AI assistance (Claude Code), directed by @abdulwahabone.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Low Risk**
> Documentation and reference-contract tests only; no runtime or API behavior changes.
> 
> **Overview**
> Closes the gap where **adapt**, **audit**, and **harden** could treat viewport screenshots as touch testing (e.g. iOS sliders that never complete a drag).
> 
> **adapt**’s verify step now requires batched checks on custom controls: complete the **primary gesture**, validate **scroll vs control** behavior, and document **what produced the evidence** (synthesized touch, engine, device) plus untested gaps.
> 
> **audit** adds a **broken touch interaction** responsive criterion with code-level tells and extends the top score to require gestures under touch.
> 
> **harden** adds **interrupted gesture** edge cases (multi-pointer, `pointercancel`, lost capture, blur), manual verify steps, optional behavioral regression tests when the runner can drive input, and the same evidence-reporting rule.
> 
> A new **skill-reference contract test** locks the required phrases in those three docs.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 6c608befbf6f96b0e4c8eb27ed80973a5f880515. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: tests/skill-reference.test.mjs [hunk 1]
```diff
@@ -67,4 +67,23 @@ describe('skill reference authoring contracts', () => {
...
+    const verifyHardening = harden.match(/## Verify Hardening\n([\s\S]*?)(?:\n## |$)/)?.[1] ?? '';
+
+    assert.match(verifyAdaptations, /\*\*Primary gesture\*\*/);
+    assert.match(verifyAdaptations, /produced the evidence/);
+    assert.match(verifyAdaptations, /verify layout, never a gesture/);
+    assert.match(verifyAdaptations, /reported gap, not a blocker/);
+    assert.match(responsive, /\*\*Broken touch interaction\*\*/);
+    assert.match(responsive, /what stayed untested/);
+    assert.match(edgeCases, /\*\*Interrupted gestures\*\*[\s\S]*works without a reload/);
+    assert.match(verifyHardening, /\*\*Interrupted gestures\*\*/);
```
[the remark this answers is not in the document]
Comment (abdulwahabone) [reply tests/skill-reference.test.mjs:90]: Pinned all three in 6c608befb: adapt's scroll-across trade (`**Scroll across it** ... without activating it`), audit's `Exercise the gesture when a browser tool can synthesize touch`, and harden's `clear the dragging state and release capture`. Each phrase is absent on main, so the test fails on main's text and passes here. (AI-assisted, directed by @abdulwahabone.)

**Review by pbakaus — APPROVED** [review]
